app/app/src/main/java/de/stustanet/stustapay/ui/NavPanel.kt

```diff
@@ -159,14 +158,14 @@ fun NavDrawer(
 private fun NavDrawerEntry(
     item: NavMenuItem,
     unreadBadgeColor: Color = Color(0xFF0FFF93),
-    nfcContext: NFCContext,
+    viewModel: NavViewModel = hiltViewModel(),
     navigateTo: (NavDest) -> Unit
 ) {
     Row(
         modifier = Modifier
             .fillMaxWidth()
             .clickable {
-                nfcContext.scanRequest!!.value = item.scanNFC
+                viewModel.scan(item.scanNFC)
```

Comment:
this seems wrong here - the navdrawer should only navigate somewhere, and scannfc should not be a property of navigation items. rather some action in the nfc scanning view should trigger the `.scan` function then.

Reply:
I think the "proper" way to do this would be to give the NfcHandler access to some sort of nav state based on which it can decide whether or not to do something. Using a button or something similar won't work because we want the app to start scanning for NFC chips as soon as the view is opened to make the payment process as quick and simple as possible

Reply:
aah, i see, you want the scanning to be active as soon as the view is active. ok, then we can leave it until we get a clearer picture :)